src/compiler/watch.ts

```diff
@@ -115,10 +122,12 @@ namespace ts {
         emit(): EmitResult;
     }
 
+    export type ReportEmitErrorSummary = (errorCount: number) => void;
+
     /**
      * Helper that emit files, report diagnostics and lists emitted and/or source files depending on compiler options
      */
-    export function emitFilesAndReportErrors(program: ProgramToEmitFilesAndReportErrors, reportDiagnostic: DiagnosticReporter, writeFileName?: (s: string) => void) {
+    export function emitFilesAndReportErrors(program: ProgramToEmitFilesAndReportErrors, reportDiagnostic: DiagnosticReporter, reportSummary?: ReportEmitErrorSummary, writeFileName?: (s: string) => void) {

```

Comment:
I think you want to change the definition to 
```typescript
export function emitFilesAndReportErrors(program: ProgramToEmitFilesAndReportErrors, reportDiagnostic: DiagnosticReporter, writeFileName: (s: string) => void, reportSummary?: ReportEmitErrorSummary) 
```
currently tsc.ts's use of emitFilesAndReportErrors is broken ()